bool retract_cmd
---
bool retract_reply